gson/src/main/java/com/google/gson/UnknownFieldStrategy.java

```diff
@@ -0,0 +1,79 @@
+package com.google.gson;
+
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import java.io.IOException;
+
+/**
+ * A strategy defining how to handle unknown fields during reflection-based deserialization.
+ *
```

Comment:
Maybe this should have a warning that users should be careful when implementing security critical validation with this, and should add extensive tests. Because if they forget to call `setUnknownFieldStrategy` or use a default `Gson` instance somewhere, then their custom strategy will not be used.